src/plugins/ClydeGPT/index.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Vencord, a modification for Discord's desktop app
+ * Copyright (c) 2022 Vendicated, Samu and contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+import { UserStore, FluxDispatcher } from "@webpack/common";
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+const runGPT = async (msg) => {
+    const message = msg.message;
+        
+    if (!message.content.includes("@Clyde")) {
+        return;
+    }
+
+    if (message.author.id == UserStore.getCurrentUser().id && !msg?.sendMessageOptions?.nonce) {
+        return;
+    }
+
+    if (message.author?.bot) {
```

Comment:
Is this necessary? We check for author ID a few lines up.

Reply:
yes, Clyde should not respond to bots

Reply:
But you're already checking if the message author equals to the current user? If I understand the code correctly